src/core/server/saved_objects/migrations/model/model.ts

```diff
@@ -72,7 +72,26 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
   if (stateP.controlState === 'INIT') {
     const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
 
-    if (Either.isRight(res)) {
+    if (Either.isLeft(res)) {
+      const left = res.left;
+      if (isLeftTypeof(left, 'cluster_routing_allocation_disabled')) {
+        return {
+          ...stateP,
+          controlState: 'FATAL',
+          reason: `Cluster routing allocation is not enabled. To proceed, please enable routing.`,
```

Comment:
Maybe we should do both? We give great guidance inline in the logs for other potential solutions. Maybe something like:
`The elasticsearch cluster has cluster routing allocation disabled. To proceed, please remove the cluster routing allocation settings with PUT /_cluster/settings {"transient": {"cluster.routing.allocation.enable": null}, "persistent": {"cluster.routing.allocation.enable": null}}. Refer to the [docs] for more details.`

However, giving the full command in the logs is tricky, and prone to error should these settings ever change (or the API changes 🤷‍♀️ ).

Reply:
> However, giving the full command in the logs is risky if these settings options change (or the API changes 🤷‍♀️ ) and we don't update the code accordingly.

This would only happen when upgrading to another major. So we would have an opportunity to make sure the command is correct in e.g. 8.last. Still there's a risk that we forget, but this scenario happens extremely rarely so the impact is fairly small.

Reply:
Adding doclinks to the saved objects service in core means we need to modify the setup and start contracts and pass it down as a dependency. I see value in adding these but think we can add that as a follow up enhancement to keep this PR relatively small. I've created an issue for that: https://github.com/elastic/kibana/issues/126864